std/algorithm/iteration.d

```diff
@@ -4852,37 +4905,101 @@ private struct UniqResult(alias pred, Range)
     void popFront()
     {
         assert(!empty, "Attempting to popFront an empty uniq.");
-        auto last = _input.front;
-        do
+        if (_input.empty)
         {
-            _input.popFront();
+            _isInFront = false;
+            static if (isBidirectionalRange!Range)
+            {
+                if (_isOverlapping)
+                {
+                    _isInBack = false;
+                }
+            }
+        }
+        else
+        {
+            _isInFront = true;
+            _front = _input.front;
```

Comment:
I don't think the impact is high, you need to make a copy every call to `popFront` anyway.